src/components/screens/DocsScreen/Feedback.tsx

```diff
@@ -0,0 +1,274 @@
+  box-shadow: ${color.border} 0 0 0 1px inset;
+
+  width: 100%;
+  min-height: 150px;
+
+  &:focus {
+    box-shadow: ${color.secondary} 0 0 0 1px inset;
+  }
+
+  &::placeholder {
+    color: ${color.mediumdark};
+  }
+`;
+
+const inaccessiblyVisuallyHidden = css`
+  clip: rect(0 0 0 0);
+  clip-path: inset(50%);
+  height: 1px;
+  overflow: hidden;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
+`;
+
+const SpuriousLabel = styled.label`
+  ${inaccessiblyVisuallyHidden}
+`;
+
+const SpuriousTextarea = styled((props) => (
+  <Textarea {...props} aria-hidden="true" tabIndex={-1} />
+))`
+  ${inaccessiblyVisuallyHidden}
+`;
+
+export const Feedback = ({
+  path,
+  version,
+  framework,
+  codeLanguage,
+  // @ts-expect-error - For Storybook only
+  forceRating = null,
+  // @ts-expect-error - For Storybook only
+  forceResultUrl = '',
+  // @ts-expect-error - For Storybook only
+  forceError = '',
+}: FeedbackProps) => {
+  const [rating, setRating] = React.useState<'up' | 'down' | null>(forceRating);
+  const [comment, setComment] = React.useState('');
+  const [spuriousComment, setSpuriousComment] = React.useState('');
+  const [error, setError] = React.useState(forceError);
+  const [resultUrl, setResultUrl] = React.useState(forceResultUrl);
+
+  const expanded = rating || resultUrl || error;
+
+  const formRef = React.useRef<HTMLFormElement | null>(null);
+  const textareaRef = React.useRef<HTMLTextAreaElement | null>(null);
+
+  const commentFieldId = React.useMemo(() => Math.random().toString(36).substring(2, 15), []);
+
+  const handleRatingClick = (whichRating) => () => {
+    setError('');
+    setRating(whichRating);
+    setTimeout(() => {
+      formRef.current?.scrollIntoView({ block: 'nearest', behavior: 'smooth' });
+      textareaRef.current?.focus();
+    }, heightTransitionTime);
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    try {
+      const response = await fetch(DOCS_FEEDBACK_URL, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          path,
+          version,
+          framework,
+          codeLanguage,
+          rating,
+          comment,
+          spuriousComment,
+        }),
+      });
+      if (response.ok) {
+        const { url } = await response.json();
+        setResultUrl(url);
+      } else {
+        throw new Error();
+      }
+    } catch (e) {
+      setError(e);
+      // eslint-disable-next-line no-console
+      console.error(e);
+    }
+
+    setRating(null);
+    setComment('');
+  };
+
+  const form = (
+    <CommentForm ref={formRef} isExpanded={expanded} onSubmit={handleSubmit}>
+      {/* eslint-disable-next-line no-nested-ternary */}
+      {error ? (
+        <OutlineCTA badge={<Badge status="error">Error</Badge>} action={<></>}>
+          Something went wrong. Please try again.
+        </OutlineCTA>
+      ) : resultUrl ? (
```

Comment:
**Nit pick**:

Anyway we can break this down? Nested ternary statements give me the brain scramblies 😵